torch/csrc/jit/source_range.h

```diff
@@ -170,6 +231,26 @@ struct CAFFE2_API SourceRange {
     return serialized_;
   }
 
+  static std::shared_ptr<SourceRange> __setstate__(const c10::IValue& iv) {
+    auto tup_elems = iv.toTuple()->elements();
+    TORCH_INTERNAL_ASSERT(tup_elems.size() == 3);
+    std::shared_ptr<Source> source_ = Source::__setstate__(tup_elems[0]);
+    int64_t start_ = tup_elems[1].toInt();
+    int64_t end_ = tup_elems[2].toInt();
+    return std::make_shared<SourceRange>(source_, start_, end_);
+  }
+
+  std::shared_ptr<SourceRange> orig_range() const {
+    // Cache this to prevent log(n) lookup every time.
```

Comment:
SourceRange objects are value objects. They are passed by value all over the place in the compiler. Trying to cache the lookup here won't work because there won't be cases where two things share the same source range. I think it would be possible to keep the cache inside the DebugInfo object, however, where it can keep a hash_table of queried byte offset -> SourceRange.